src/geometry/adjoints.jl

```diff
@@ -50,6 +50,24 @@ function ChainRulesCore.rrule(::Type{SVector{N, T}}, x...) where {N, T}
     return y, svector_const_pb
 end
 
+function ChainRulesCore.rrule(::Type{SMatrix{N, M, T, K}}, x...) where {N, M, T, K}
```

Comment:
I believe this method can be absorbed into the previous `SVector` version via something like

```julia
function ChainRulesCore.rrule(::Type{AT}, x...) where {AT<:SArray}
    y = AT(x...)
    ...
end
```

Reply:
Yes that seems to be a good idea. I'll make it to the type `::Type{SArray{...}}` only as all `SVector`s and `SMatrix`es can be represented using `SArray`. We can represent higher dimensional Arrays too using the `SArray` interface.

Reply:
What happens if you nograd this kind of rrule entirely?

Reply:
That wouldn't be a good idea. We use StaticArrays widely in intermediate representation, and that would surely cause gradients to stop propagating.
Let me try that once.